pydantic/types.py

```diff
@@ -3007,3 +3008,31 @@ def __get_pydantic_core_schema__(cls, source_type: Any, handler: GetCoreSchemaHa
 That is, instead of a [`ValidationError`][pydantic_core.ValidationError] being propagated up and the entire iterable being discarded
 any invalid items are discarded and the valid ones are returned.
 """
+
+
+@_dataclasses.dataclass(**_internal_dataclass.slots_true, frozen=True)
+class FailFast:
+    """A `FailFast` can be used to specify that validation should stop at the first error.
+
+    This can be useful when you want to validate a large amount of data and you only need to know if it's valid or not.
+
+    ```py
+    from typing import List
+    from typing_extensions import Annotated
+    from pydantic import BaseModel, FailFast
+
+    class Model(BaseModel):
+        x: Annotated[List[int], FailFast()]
+
+    # This will raise a single error for the first invalid value and stop validation
+    obj = Model(x=[1, 2, 'a', 4, 5, 'b', 7, 8, 9, 'c'])
```

Comment:
I believe there's a test failing here just bc of this formatting - example looks good overall.

You can also use `mkdocs` [annotations](https://squidfunk.github.io/mkdocs-material/reference/annotations/) to make this look a bit cleaner.

Reply:
I guess that's because this version of `pydantic-core` doesn't support `fail_fast` and raises `SchemaError`

Reply:
Ah yes, you're right. Should be fixed once we bump `pydantic-core` with your new feature.